Explorer/Assets/DCL/Multiplayer/Profiles/Tables/EntityParticipantTable.cs

```diff
@@ -0,0 +1,37 @@
+using Arch.Core;
+using DCL.Utilities.Extensions;
+using System.Collections.Generic;
+
+namespace DCL.Multiplayer.Profiles.Tables
+{
+    /// <summary>
+    ///     NOT Thread-safe
```

Comment:
Did you try [ConcurrentDictionary](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-8.0)?